shpkpr/marathon/validate.py

```diff
@@ -25,18 +26,32 @@ def format_message(self):
 
 class SchemaValidator(object):
 
-    def __init__(self, schema_path):
+    def __init__(self, schema_path, strip=False):
         self._schema = self._read_schema_from_file(schema_path)
+        self._strip = strip
 
     @exceptions.rewrap(jsonschema.ValidationError, ValidationError)
     def __call__(self, application):
         jsonschema.validate(application, self._schema)
+        if not self._strip:
+            return application
+        return self.strip(application)
 
     def _read_schema_from_file(self, path):
         with open(path, 'r') as f:
             schema = f.read()
         return json.loads(schema)
 
+    def strip(self, application):
+        """Removes any top level properties from the application dictionary
+        that haven't been validated by the schema. This helps prevent the
+        unintended use of unvalidated properties.
+        """
+        stripped_app = {}
+        for key in self._schema['required']:
+            stripped_app[key] = copy.deepcopy(application[key])
+        return stripped_app
 
-validate_app = SchemaValidator(load_schema("app"))
+
+validate_app = SchemaValidator(load_schema("app"), strip=True)
```

Comment:
It would probably be better to not have the validator mutate the application. Validation seems more like a true / false action and reducing the schema down something that would live in the client which handles the initial response before validating it.